src/target/riscv/riscv-013.c

```diff
@@ -1512,22 +1505,19 @@ static int write_memory(struct target *target, target_addr_t address,
 
 	select_dmi(target);
 
-	/* This program uses two temporary registers.  A word of data and the
```

Comment:
why did you delete all this commentary? Is it no longer true?

Reply:
It's partly true. There are some comments below which explain a bit more what's going on. (It's easier to read not in diff form, since basically the function was completely rewritten.)